<?xml version="1.0"?>

<launch>
  <env name="GAZEBO_RESOURCE_PATH" value="$(find amr_gazebo)/media/materials"/>

  <include file="$(find gazebo_ros)/launch/empty_world.launch">
<? ignore
    <arg name="world_name" value="$(find amr_gazebo)/worlds/factory_sim.world"/>
?>
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="true"/>
    <arg name="recording" value="false"/>
    <arg name="debug" value="false"/>
    <arg name="verbose" value="true"/>
  </include>

<arg name="shelft_type" value="2in1"/>

<!-- Spawn 2-in-1 in Gazebo -->
<group ns="$(arg shelft_type)">
    <arg name="robot_name" value="$(arg shelft_type)"/>
    <param name="tf_prefix" value="$(arg robot_name)" />  
    <!-- interpret xacro into urdf-->
    <param name="robot_description" command="$(find xacro)/xacro --inorder $(find amr_description)/urdf/$(arg shelft_type).urdf.xacro robotName:=$(arg robot_name)" />
    <!-- spawn model from parameter server : robot_description -->
    <node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf_$(arg robot_name)" args="-urdf -model $(arg robot_name) -x 0 -y 1 -z 0 -param robot_description" />
</group>



<?ignore
<!-- Spawn solamr_1 in Gazebo -->
<group ns="solamr_1">
    <arg name="robot_name" value="solamr_1"/>
    <param name="tf_prefix" value="$(arg robot_name)" />  
    <!-- interpret xacro into urdf-->
    <param name="robot_description" command="$(find xacro)/xacro --inorder $(find amr_description)/urdf/solamr_rod.urdf.xacro robotName:=$(arg robot_name)" />
    <!-- spawn model from parameter server : robot_description -->
    <node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf_$(arg robot_name)" args="-urdf -model $(arg robot_name) -x -2 -y 1 -z 0 -param robot_description" />
    <!-- control file: controller spawner, robot_state_publisher -->
    <include file="$(find amr_control)/launch/amr_control.launch">
        <arg name="robot_ns" value="$(arg robot_name)" />
    </include>
</group>
?>


<!-- keyboard teleop 
    <node pkg="amr_gazebo" type="teleop_key.py" name="solamr_teleop" output="screen">
    </node>
-->

</launch>
